services/net/image/ImageAction.cs

```diff
@@ -100,14 +100,15 @@ public ImageAction(IApiService api, IOptions<ImageOptions> options, ILogger<Imag
                 {
                     // If another process has it in progress only attempt to do an import if it's
                     // more than an 5 minutes old. Assumption is that it is stuck.
-                    await this.UpdateContentReferenceAsync(reference, WorkflowStatus.InProgress);
+                    reference = await UpdateContentReferenceAsync(reference, WorkflowStatus.InProgress);
                 }
                 else reference = null;
 
                 if (reference != null)
                 {
                     await CopyImageAsync(client, manager.Ingest, remotePath.CombineWith(file.Name));
-                    await this.ContentReceivedAsync(manager, reference, CreateSourceContent(manager.Ingest, reference));
+                    reference = await FindContentReferenceAsync(reference.Source, reference.Uid);
```

Comment:
What's the value in this?  You already have an up-to-date reference (line 94).  If it enters the if statements it will get a updated reference from (line 97, or 103).  You don't need to fetch again.

Reply:
Removed, however, depends on how long it will take for CopyImageAsync to run, this could be a potential cause I guess? Thanks